bioconda_utils/lint_functions.py

```diff
@@ -185,6 +203,22 @@ def has_windows_bat_file(recipe, meta, df):
             'fix': 'remove windows .bat files'
         }
 
+
+def should_be_noarch(recipe, meta, df):
+    deps = _get_deps(meta)
+    if (
+        ('gcc' not in deps) and
+        ('python' in deps) and
+        not _has_preprocessing_selector(meta)
```

Comment:
What is the point of the preprocessing selector? Is it just because we cannot be sure in that case?